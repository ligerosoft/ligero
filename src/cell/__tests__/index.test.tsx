import { RightOutlined } from '@ant-design/icons';
import { mount } from 'enzyme';
import React from 'react';
import Cell from '..';

describe('Cell', () => {
  it('render cell correctly', () => {
    const wrapper = mount(<Cell>cell</Cell>);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with ellipsis', () => {
    const wrapper = mount(
      <Cell ellipsis>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit. Velit natus enim dolorum autem ad
        aspernatur vitae, eveniet blanditiis consequatur dignissimos iste quibusdam doloribus
        nostrum voluptates provident sit repudiandae, nam culpa.
      </Cell>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render with align correctly', () => {
    const wrapper = mount(<Cell />);
    const aligns = ['top', 'center', 'bottom'];
    aligns.forEach(align => {
      wrapper.setProps({ align });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('render with label correctly', () => {
    const wrapper = mount(<Cell label="label">content</Cell>);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with extra icon correctly', () => {
    const wrapper = mount(<Cell extra={<RightOutlined />}>content</Cell>);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with right and extra', () => {
    const wrapper = mount(
      <Cell right="right" extra={<RightOutlined />}>
        content
      </Cell>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render cell group correctly', () => {
    const wrapper = mount(
      <Cell.Group title="组一">
        <Cell label="描述信息">单元格</Cell>
        <Cell label="描述信息">单元格</Cell>
      </Cell.Group>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
