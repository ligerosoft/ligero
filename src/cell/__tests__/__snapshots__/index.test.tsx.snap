// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell render cell correctly 1`] = `
<@ligero/cell>
  <div
    className="ligero-cell"
    role="button"
  >
    <div
      className="ligero-cell__content"
    >
      cell
    </div>
  </div>
</@ligero/cell>
`;

exports[`Cell render cell group correctly 1`] = `
<@ligero/cell/group
  title="组一"
>
  <div
    className="ligero-cell-group__title"
  >
    组一
  </div>
  <div
    className="ligero-cell-group ligero-hairline--top-bottom"
  >
    <@ligero/cell
      label="描述信息"
    >
      <div
        className="ligero-cell"
        role="button"
      >
        <div
          className="ligero-cell__content"
        >
          单元格
          <div
            className="ligero-cell__label"
          >
            描述信息
          </div>
        </div>
      </div>
    </@ligero/cell>
    <@ligero/cell
      label="描述信息"
    >
      <div
        className="ligero-cell"
        role="button"
      >
        <div
          className="ligero-cell__content"
        >
          单元格
          <div
            className="ligero-cell__label"
          >
            描述信息
          </div>
        </div>
      </div>
    </@ligero/cell>
  </div>
</@ligero/cell/group>
`;

exports[`Cell render with align correctly 1`] = `
<@ligero/cell
  align="top"
>
  <div
    className="ligero-cell ligero-cell--top"
    role="button"
  >
    <div
      className="ligero-cell__content"
    />
  </div>
</@ligero/cell>
`;

exports[`Cell render with align correctly 2`] = `
<@ligero/cell
  align="center"
>
  <div
    className="ligero-cell ligero-cell--center"
    role="button"
  >
    <div
      className="ligero-cell__content"
    />
  </div>
</@ligero/cell>
`;

exports[`Cell render with align correctly 3`] = `
<@ligero/cell
  align="bottom"
>
  <div
    className="ligero-cell ligero-cell--bottom"
    role="button"
  >
    <div
      className="ligero-cell__content"
    />
  </div>
</@ligero/cell>
`;

exports[`Cell render with ellipsis 1`] = `
<@ligero/cell
  ellipsis={true}
>
  <div
    className="ligero-cell"
    role="button"
  >
    <div
      className="ligero-cell__content ligero-cell__content--ellipsis"
    >
      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Velit natus enim dolorum autem ad aspernatur vitae, eveniet blanditiis consequatur dignissimos iste quibusdam doloribus nostrum voluptates provident sit repudiandae, nam culpa.
    </div>
  </div>
</@ligero/cell>
`;

exports[`Cell render with extra icon correctly 1`] = `
<@ligero/cell
  extra={<ForwardRef(RightOutlined) />}
>
  <div
    className="ligero-cell"
    role="button"
  >
    <div
      className="ligero-cell__content"
    >
      content
    </div>
    <div
      className="ligero-cell__extra"
    >
      <ForwardRef(RightOutlined)>
        <AntdIcon
          icon={
            Object {
              "icon": Object {
                "attrs": Object {
                  "focusable": "false",
                  "viewBox": "64 64 896 896",
                },
                "children": Array [
                  Object {
                    "attrs": Object {
                      "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                    },
                    "tag": "path",
                  },
                ],
                "tag": "svg",
              },
              "name": "right",
              "theme": "outlined",
            }
          }
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <IconReact
              icon={
                Object {
                  "icon": Object {
                    "attrs": Object {
                      "focusable": "false",
                      "viewBox": "64 64 896 896",
                    },
                    "children": Array [
                      Object {
                        "attrs": Object {
                          "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                        },
                        "tag": "path",
                      },
                    ],
                    "tag": "svg",
                  },
                  "name": "right",
                  "theme": "outlined",
                }
              }
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                key="svg-right"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  key="svg-right-svg-0"
                />
              </svg>
            </IconReact>
          </span>
        </AntdIcon>
      </ForwardRef(RightOutlined)>
    </div>
  </div>
</@ligero/cell>
`;

exports[`Cell render with label correctly 1`] = `
<@ligero/cell
  label="label"
>
  <div
    className="ligero-cell"
    role="button"
  >
    <div
      className="ligero-cell__content"
    >
      content
      <div
        className="ligero-cell__label"
      >
        label
      </div>
    </div>
  </div>
</@ligero/cell>
`;

exports[`Cell render with right and extra 1`] = `
<@ligero/cell
  extra={<ForwardRef(RightOutlined) />}
  right="right"
>
  <div
    className="ligero-cell"
    role="button"
  >
    <div
      className="ligero-cell__content"
    >
      content
    </div>
    <div
      className="ligero-cell__right"
    >
      right
    </div>
    <div
      className="ligero-cell__extra"
    >
      <ForwardRef(RightOutlined)>
        <AntdIcon
          icon={
            Object {
              "icon": Object {
                "attrs": Object {
                  "focusable": "false",
                  "viewBox": "64 64 896 896",
                },
                "children": Array [
                  Object {
                    "attrs": Object {
                      "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                    },
                    "tag": "path",
                  },
                ],
                "tag": "svg",
              },
              "name": "right",
              "theme": "outlined",
            }
          }
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <IconReact
              icon={
                Object {
                  "icon": Object {
                    "attrs": Object {
                      "focusable": "false",
                      "viewBox": "64 64 896 896",
                    },
                    "children": Array [
                      Object {
                        "attrs": Object {
                          "d": "M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z",
                        },
                        "tag": "path",
                      },
                    ],
                    "tag": "svg",
                  },
                  "name": "right",
                  "theme": "outlined",
                }
              }
            >
              <svg
                aria-hidden="true"
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                key="svg-right"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  key="svg-right-svg-0"
                />
              </svg>
            </IconReact>
          </span>
        </AntdIcon>
      </ForwardRef(RightOutlined)>
    </div>
  </div>
</@ligero/cell>
`;
